src/lib/unpackTags.ts

```diff
@@ -0,0 +1,64 @@
+import { Tag } from 'alks.js';
+/**
+ * Parse tags into a Tags object from either JSON or the AWS defined shorthand for tag options
+ *
+ * @param inputs - a list of strings containing key-value pairs in one of two forms consistent with the AWS CLI
+ * @see https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-shorthand.html
+ *
+ * @example
+ * parseKeyValuePairs(['Key=key1,Value=val1'])
+ *
+ * @example
+ * parseKeyValuePairs(['{"Key":"key1","Value":"value1"}'])
+ */
+
+export function unpackTags(inputs: string[]): Tag[] {
+  const tags: Tag[] = [];
+
+  for (const input of inputs) {
+    let record: Record<string, string> | undefined = parseShorthand(input);
+    if (typeof record == 'undefined') {
+      let obj = JSON.parse(input, (_, value) =>
+        typeof value !== 'object' ? String(value) : value
+      );
+      // iterable object must be a list otherwise the length returns undefined
+      obj = obj.length ? obj : [obj];
+
+      record = {} as Record<string, string>;
+      for (const entry of obj) {
+        record[entry.Key] = entry.Value;
+      }
+    }
+    for (const [key, value] of Object.entries(record)) {
+      const t: Tag = {
+        key,
+        value,
+      };
+      tags.push(t);
+    }
+  }
+  return tags;
+}
+
+function parseShorthand(input: string): Record<string, string> | undefined {
+  const record: Record<string, string> = {};
+  try {
+    JSON.parse(input);
+    return;
+  } catch (e) {
+    const errorMsg =
+      'Improper tag syntax. Should look like either \'{"Key":"key1", "Value":"val1"}\' or "Key=key1,Value=val1"';
+    try {
+      const pair = input.split(',');
+      const key = pair[0].split('=');
+      const value = pair[1].split('=');
+      if (!(key[0] === 'Key' && value[0] === 'Value')) {
+        throw SyntaxError(errorMsg);
+      }
+      record[key[1]] = value[1];
+    } catch (e) {
+      throw SyntaxError(errorMsg);
+    }
+  }
+  return record;
+}
```

Comment:
Considering that it's the end of the sprint and that you already have working tests around this, I'll let this slide but I would normally recommend breaking this function out into its own file and giving it its own set of unit tests. I don't expect we'll really end up reusing this functions so it's not for that, but giving this function its own tests and mocking it out in the tests for the function that uses it makes the unit of test smaller, which makes it easier to maintain since any failed unit tests in the future will only point to one of these functions rather than both, reducing the amount of logic that might contain the bug and making it much quicker to diagnose the cause.

Like I said though, something to consider for next time but don't feel obligated to fix it now if time doesn't permit